### ISSUE TITLE
OTA example doesn't consider the watchdog

### ISSUE THREAD

**Issue by hajosc** [espressif/arduino-esp32#10707]
### Board

all ESP32

### Device Description

Any ESP32

### Hardware Configuration

n/a

### Version

latest master (checkout manually)

### IDE Name

Arduino IDE

### Operating System

any

### Flash frequency

any

### PSRAM enabled

yes

### Upload speed

OTA

### Description

A lot of people across the Internet complain about ArduinoOTA failing with timeouts. They reduce code sizes, increase transfer speeds over WiFi and have random success thereafter. I believe one major reason is the watchdog biting them. 
I suggest to add esp_task_wdt_reset(); to ArduinoOTA.onProgress() in the example.

### Sketch

```cpp
ArduinoOTA.onProgress([](unsigned int progress, unsigned int total) {
			esp_task_wdt_reset(); // reset watchdog
		});
```

### Debug Message

```plain
.
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by robertlipe** [comment]
It's probably not a _bad_ idea.

To make it easier to help people to help you @hajosc , it would have been helpful to include permalinks to the exact code you were suggesting mods to. It's probably some subset of:

https://github.com/espressif/arduino-esp32/blob/6015fd73e0d9d065c80fc79ef0f35d3273a217de/libraries/ArduinoOTA/examples/BasicOTA/BasicOTA.ino#L62
https://github.com/espressif/arduino-esp32/blob/6015fd73e0d9d065c80fc79ef0f35d3273a217de/libraries/ArduinoOTA/src/ArduinoOTA.cpp#L46
https://github.com/espressif/arduino-esp32/blob/6015fd73e0d9d065c80fc79ef0f35d3273a217de/libraries/HTTPUpdate/examples/httpUpdate/httpUpdate.ino#L46

I'll note that as of 2024-07-18 that none of the three files have made this. If a maintainer wants this and thinks it's easier for me to paste in those three lines, completely blindly and untested, than to accept an actually tested PR from OP or to develop a path themselves, I'm happy enough to do so.

The limit of my help is at being a keyboard monkey, not in setting up a test fixture. This is why we have CI. :-)

Signed,
Fly By PR reviewer.

**Comment by hajosc** [comment]
You are absolutely correct @robertlipe -- I completely missed the fact that there's more than one example, apologies for that... I in fact referred to 

https://github.com/espressif/arduino-esp32/blob/6015fd73e0d9d065c80fc79ef0f35d3273a217de/libraries/ArduinoOTA/examples/BasicOTA/BasicOTA.ino#L62 

but it likely affects the others you mentioned above equally. I haven't tested those though. Note that for the error to occur an OTA update obviously must be large enough, causing the WD timeout to be able to trigger.

**Comment by robertlipe** [comment]
That's why I was gently pushing back on you to either actually submit a PR with the fix(es) applied since you apparently had an environment where this mattered. (I'm bewildered it doesn't matter in MOST environments. A download takes, what, 30 seconds? Surely watchpuppy needs to be fed by then! Or maybe it actually doesn't and this whole line of patching is unnecessary. I sort of understand the envelope around this line of questioning, but I don't have code that triggers it, so I'm not terribly happy to pound my fist on a desk and _demand_ that a 3-line PR be applied just because I can't vouch for it fixing a problem I don't have.

I'd suggest kicking this up to a PR. 

My only hesitancy around this not being an "obviously required" fix is that projects all over the world aren't bursting into flames. My guess is that this is usually called from within a task and the watchdog is being reset on a task switch so it never _actually_ expires.

Having an absurd number of OTAs and eamples was something I fussed about in another issue I just pencil-whipped minutes away from reviewing this one. How many is too many?

**Comment by hajosc** [comment]
Wow, why do Software things have to become personal? I have no access to a PC, let alone an embedded development platform for the next few months due to extensive travel. Back then I wanted to help others with findings I made, apologies if this is not appreciated. Further, I have honestly not the slightest idea what a PR or CI are - you are far ahead on the learning curve there -  and yes, I might be totally wrong in what fixed a problem for me. Also, I fully acknowledge that I as non-native English speaker will never be able to match your choice of words; probably never comprehending what "pencil-whipped minutes" are. Please do continue being a marvelous keyboard monkey, I'm meanwhile out of this for good.
